### DISCUSSION TITLE
分批查询全量数据

### DISCUSSION THREAD

**Discussion by silear** [milvus-io/milvus#38075] [Q&A and General discussion]
1、pymilvus怎么分批次查询全量数据，使用limit和offset吗，但是并没有生效，请问具体要怎么使用呢？
2、为什么我的limit设置并没有生效，设置limit=10，但是查出来是7个结果

**Comment by yhmo** [comment]
query全量数据可以使用QueryIterator: https://milvus.io/docs/get-and-scalar-query.md#Use-QueryIterator%E2%80%8B
这个工具已经帮你做好了分批查询，它内部使用的方法主要是range search: https://milvus.io/docs/range-search.md#Range-Search%E2%80%8B
